features/admin.verifiable-credentials.v1/components/vc-template-list.tsx

```diff
@@ -0,0 +1,278 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getEmptyPlaceholderIllustrations } from "@wso2is/admin.core.v1/configs/ui";
+import { AppConstants } from "@wso2is/admin.core.v1/constants/app-constants";
+import { history } from "@wso2is/admin.core.v1/helpers/history";
+import { AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import {
+    AnimatedAvatar,
+    AppAvatar,
+    ConfirmationModal,
+    DataTable,
+    EmptyPlaceholder,
+    PrimaryButton,
+    TableActionsInterface,
+    TableColumnInterface
+} from "@wso2is/react-components";
+import React, { ReactElement, ReactNode, SyntheticEvent, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Header, Icon, SemanticICONS } from "semantic-ui-react";
+import { deleteVCTemplate } from "../api/verifiable-credentials";
+import { VCTemplateListItem } from "../models/verifiable-credentials";
+
+interface VCTemplateListProps extends IdentifiableComponentInterface {
+    advancedSearch: ReactNode;
+    isLoading: boolean;
+    mutateTemplateList: () => void;
+    list: VCTemplateListItem[];
+    onSearchQueryClear?: () => void;
+    searchQuery?: string;
+    setShowAddTemplateWizard: () => void;
+}
+
+/**
+ * Verifiable Credentials template list component.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const VCTemplateList = ({
+    advancedSearch,
+    isLoading,
+    list,
+    mutateTemplateList,
+    onSearchQueryClear: _onSearchQueryClear,
+    searchQuery: _searchQuery,
+    setShowAddTemplateWizard,
+    ["data-componentid"]: componentId
+}: VCTemplateListProps): ReactElement => {
+    const dispatch: any = useDispatch();
```

Comment:
Let's use type `Dispatch` from `redux`.

```suggestion
    const dispatch: Dispatch = useDispatch();
```